### ISSUE TITLE
Adding option to set instrument names to a different language than the rest of the app

### ISSUE THREAD

**Issue by Yanko64** [musescore/MuseScore#19071]
### Your idea

the app should have the option to keep instrument names in English (or change them to another language) when using a translated version of the app.

### Problem to be solved

this will make it more comfortable for non-English speakers to use the app translated into their language while also keeping our scores looking professional by default without having to change instrument names manually, which can save some time, especially with scores containing a lot of instruments. 

### Additional context

this was briefly discussed in the Discord server:
- [here](https://discord.com/channels/818804595450445834/818852942882275379/1126919588790149151) 
- [and here](https://discord.com/channels/818804595450445834/818852942882275379/1141267206592790620)

another suggestion that is partly related to this one is always keeping expression text in its original language. It was also discussed in the Discord server at a similar time [here](https://discord.com/channels/818804595450445834/818852942882275379/1128665646427816048) and reported (by me) here #18560

**Comment by bkunda** [comment]
@shoogle FYI

**Comment by cbjeukendrup** [comment]
See also my comment at https://github.com/musescore/MuseScore/issues/19154#issuecomment-1710858286, which includes some initial thoughts about implementation.
I also thought that this might be a nice GSoC project, although it might turn out that it's not big enough or that we want to do this ourselves.

**Comment by shoogle** [comment]
It looks like it will take some trickery to get Qt to return translations in multiple languages at once. The documentation for [QApplication::installTranslator](https://doc.qt.io/qt-6/qcoreapplication.html#installTranslator) says:

> Translations are searched for in the reverse order in which they were installed [...] The search stops as soon as a translation containing a matching string is found.

 So in theory we could just install a translator for the UI in one language and a translator for instrument names in a different language. However, that won't work if we want instrument _descriptions_ to be in the UI language, and there might be some places where we want to see the name in the UI language as well.

![image](https://github.com/musescore/MuseScore/assets/11011881/7b0c4985-8975-442d-91b9-cd1664852f4c)

When we want to see a "foreign" instrument name, we may have to work with the [QTranslator](https://doc.qt.io/qt-6/qtranslator.html) class directly and call [its translate function](https://doc.qt.io/qt-6/qtranslator.html#translate) rather than [QApplication::translate](https://doc.qt.io/qt-6/qcoreapplication.html#translate).

If we want the translations to work without Qt then we may end up writing our own parser for `.qm` binary translation files, which would be a whole different ball game!

**Comment by cbjeukendrup** [comment]
Yes, I think we should use QTranslator directly, but hide it inside the implementation of LocalizationProvider. We could do something similar like we did with `Painter`, namely creating some sort of interface class that allows for multiple backends. So we use a QTranslator based backend, while some other project that wants to utilise our engraving system might implement its own backend.

**Comment by shoogle** [comment]
Some reminders for later, it may be useful for users to be able to:

1. Choose a different language for _instrument names_ compared to the _UI_.
    - E.g. UI in English, instrument names in Italian.
2. Customize where traits (transposition, tuning, string count, etc.) appear within instrument names.
    - E.g. "_C Trumpet_" vs. "_Trumpet in C_"
3. Choose a different language/convention for how traits are displayed.
    - E.g. "_7-course_" vs. "_7-string_"
    - E.g. Italian _instrument names_ but with German _note names_ ("_Clarinetto in H_").
    - Apparently a useful option for Chinese is to put accidentals before or after the note (e.g. `♭B` vs `B♭`). See [Discord](https://discord.com/channels/818804595450445834/818852942882275379/1318006666561716377).
4. Change all these settings after a score is created.
    - E.g. if I find a score with Chinese instrument names, I might want to change them to English.
    - This could be done by storing placeholders in MSCX (e.g. "_{instrument} in {transposition}_") and then constructing instrument names dynamically. Users would still need the ability to edit names manually for edge cases.
    - Alternatively, names can be baked-in as strings as they are now, but if the user switches language then it resets all instrument names to the default values in the new language.
